
// Generated by hack/generate-e2e-test.py
package test.e2e.core.function;

import static org.junit.Assert.fail;
import org.junit.Test;
import test.e2e.core.E2eTestBase;

public class CurveDetail1Test extends E2eTestBase {

  public CurveDetail1Test() {
    coreCodeTemplate = "settings <- function() {\n    size(100, 100, P3D)\n}\n\nsetup <- function() {\n    noFill()\n    noLoop()\n}\n\ndraw <- function() {\n    curveDetail(1)\n    drawCurves(-15)\n    stroke(126)\n    curveDetail(2)\n    drawCurves(0)\n    stroke(255)\n    curveDetail(4)\n    drawCurves(15)\n}\n\ndrawCurves <- function(y) {\n    curve(5, 28 + y, 5, 28 + y, 73, 26 + y, 73, 63 + y)\n    curve(5, 28 + y, 73, 26 + y, 73, 63 + y, 15, 67 + y)\n    curve(73, 26 + y, 73, 63 + y, 15, 67 + y, 15, 67 + y)\n    saveFrame(\"%s\")\n    exit()\n}\n";
    referenceURI = "https://processing.org/reference/images/curveDetail_.png";
  }

  @Test
  public void test() {
    try {
      defaultOperation();
    } catch (Exception exception) {
      System.err.println(exception);
      fail("Should not have thrown any exception");
    }
  }
}
